### PR TITLE
JwtGenerator changes according to new UserRole logic

### LINKED ISSUE #440: Implement bit field with enumeration for Roles
Connect with fron end beforehand

### PR DESCRIPTION
**Pull request by DemetriosKoras**
All of JWT logic moved to generator class from controller. Also added more correct display of user roles in response as well as multiple tokens in headers.

### CODE REVIEW

File: CharlieBackend.Business/Helpers/JWTGenerator.cs [hunk 1]
```diff
@@ -19,15 +18,38 @@ public JwtGenerator(IOptions<AuthOptions> authOptions)
         {
             _authOptions = authOptions.Value;
         }
+        public Dictionary<string, string> ReturnJwtDictionary(AccountDto account)
```
Comment (andrii-developer) [inline CharlieBackend.Business/Helpers/JWTGenerator.cs]: Please find a more flexible solution for that. 
Use this link - https://stackoverflow.com/questions/972307/how-to-loop-through-all-enum-values-in-c

File: CharlieBackend.Business/Helpers/JWTGenerator.cs [hunk 2]
```diff
@@ -19,15 +18,38 @@ public JwtGenerator(IOptions<AuthOptions> authOptions)
...
+            if (account.Role.HasFlag(UserRole.Admin))
+            {
+                jwtDictionary.Add(UserRole.Admin.ToString(), GenerateEncodedJwt(account, UserRole.Admin));
+            }
+            
+            return jwtDictionary;
+        }
 
-        public string GenerateEncodedJwt(AccountDto account, UserRole role)
+        private string GenerateEncodedJwt(AccountDto account, UserRole role)
```
Comment (andrii-developer) [inline CharlieBackend.Business/Helpers/JWTGenerator.cs]: I think you may leave it public.

File: CharlieBackend.Api/Controllers/AccountsController.cs [hunk 3]
```diff
@@ -126,31 +112,18 @@ public async Task<ActionResult> SignIn([FromBody] AuthenticationDto authenticati
...
             {
                 first_name = foundAccount.FirstName,
                 last_name = foundAccount.LastName,
-                role = foundAccount.Role,
-                role_list = userRoleList
+                role = foundAccount.Role.ToString()
             };
-
-            Response.Headers.Add("Authorization", authorization);
+            foreach (KeyValuePair<string, string> record in userRoleToJwtToken)
```
Comment (andrii-developer) [inline CharlieBackend.Api/Controllers/AccountsController.cs]: If I remember, it was decided to put tokens into the response body, but not in headers.

File: CharlieBackend.Business/Helpers/JWTGenerator.cs (hunk 1 above)
Comment (andrii-developer) [reply CharlieBackend.Business/Helpers/JWTGenerator.cs]: It think it would be better if you name it GetRoleJwtDictionary(AccountDto account).

**Review by andrii-developer — CHANGES_REQUESTED** [review]

File: CharlieBackend.Business/Helpers/JWTGenerator.cs (hunk 2 above)
Comment (DemetriosKoras) [reply CharlieBackend.Business/Helpers/JWTGenerator.cs]: Ok!

File: CharlieBackend.Business/Helpers/JWTGenerator.cs (hunk 1 above)
Comment (DemetriosKoras) [reply CharlieBackend.Business/Helpers/JWTGenerator.cs]: Done! e818162

File: CharlieBackend.Api/Controllers/AccountsController.cs (hunk 3 above)
Comment (DemetriosKoras) [reply CharlieBackend.Api/Controllers/AccountsController.cs]: Ok, I've changed it towards more like it was, but I think this matter requires further discussion. We need to put all the tokens we have in body of response, that's clear. But do we have to leave at least one token in header? If so I've already done it. If we don't - I can remove this feature.

**Pull request merged**